dipy/reconst/dti.py

```diff
@@ -1313,25 +1378,43 @@ def wrapped_fit_tensor(design_matrix, data, return_S0_hat=False,
                                   return_S0_hat=return_S0_hat,
                                   *args, **kwargs)
             data = data.reshape(-1, data.shape[-1])
-            dtiparams = np.empty((size, 12), dtype=np.float64)
+            if 'return_lower_triangular' in kwargs:
+                sz = 7 if kwargs['return_lower_triangular'] else 12
+            else:
+                sz = 12
```

Comment:
```suggestion
sz = 7 if kwargs.get('return_lower_triangular', False) else 12
```

Reply:
Thanks, I can't commit from here due to the indent, but this makes sense (and I didn't know we could do this, it's very useful!)